### DISCUSSION TITLE
How to use it on linux

### DISCUSSION THREAD

**Discussion by zara206** [qdm12/gluetun#865] [Ideas]
I'm confused, I set up my windscribe container using this, how do I configured it so I can use it on my client? 
after executing I get nothing in `docker ps`, [Connect-a-container-to-gluetun](https://github.com/qdm12/gluetun/wiki/Connect-a-container-to-gluetun) doesn't give much information and adding `--network=container:gluetun` cause docker to throw error

<details>
  <summary>
docker run -it --rm --cap-add=NET_ADMIN -e VPNSP=windscribe -e VPN_TYPE=openvpn -e OPENVPN_USER=myuser -e OPENVPN_PASSWORD=mypass -e REGION=Netherlands qmcgaw/gluetun
  </summary>
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2022-02-20T21:30:49.183Z (commit e756273)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2022/02/22 04:24:51 WARN You are using the old environment variable VPNSP, please consider changing it to VPN_SERVICE_PROVIDER
2022/02/22 04:24:51 WARN You are using the old environment variable REGION, please consider changing it to SERVER_REGIONS
2022/02/22 04:24:51 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2022/02/22 04:24:51 INFO routing: local ethernet link found: eth0
2022/02/22 04:24:51 INFO routing: local ipnet found: 172.17.0.0/16
2022/02/22 04:24:51 INFO firewall: enabling...
2022/02/22 04:24:51 INFO firewall: enabled successfully
2022/02/22 04:24:51 INFO storage: creating /gluetun/servers.json with 11096 hardcoded servers
2022/02/22 04:24:51 INFO Alpine version: 3.15.0
2022/02/22 04:24:51 INFO OpenVPN 2.4 version: 2.4.11
2022/02/22 04:24:51 INFO OpenVPN 2.5 version: 2.5.4
2022/02/22 04:24:51 INFO Unbound version: 1.13.2
2022/02/22 04:24:51 INFO IPtables version: v1.8.7
2022/02/22 04:24:51 INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: windscribe
|   |   └── Server selection settings:
|   |       ├── VPN type: openvpn
|   |       ├── Regions: netherlands
|   |       └── OpenVPN server selection settings:
|   |           └── Protocol: UDP
|   └── OpenVPN settings:
|       ├── OpenVPN version: 2.5
|       ├── User: [set]
|       ├── Password: [set]
|       ├── Tunnel IPv6: no
|       ├── Network interface: tun0
|       ├── Run OpenVPN as: root
|       └── Verbosity level: 1
├── DNS settings:
|   ├── DNS server address to use: 127.0.0.1
|   ├── Keep existing nameserver(s): no
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Unbound settings:
|       |   ├── Authoritative servers:
|       |   |   └── cloudflare
|       |   ├── Caching: yes
|       |   ├── IPv6: no
|       |   ├── Verbosity level: 1
|       |   ├── Verbosity details level: 0
|       |   ├── Validation log level: 0
|       |   ├── System user: root
|       |   └── Allowed networks:
|       |       ├── 0.0.0.0/0
|       |       └── ::/0
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:7f00:1/104
|               ├── ::ffff:a00:0/104
|               ├── ::ffff:a9fe:0/112
|               ├── ::ffff:ac10:0/108
|               └── ::ffff:c0a8:0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Address to ping: github.com
|   └── VPN wait durations:
|       ├── Initial duration: 5s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   └── Process GID: 1000
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2022/02/22 04:24:51 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2022/02/22 04:24:51 INFO routing: adding route for 0.0.0.0/0
2022/02/22 04:24:51 INFO firewall: setting allowed subnets...
2022/02/22 04:24:51 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2022/02/22 04:24:51 INFO TUN device is not available: open /dev/net/tun: no such file or directory; creating it...
2022/02/22 04:24:51 INFO routing cleanup...
2022/02/22 04:24:51 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2022/02/22 04:24:51 INFO routing: deleting route for 0.0.0.0/0
2022/02/22 04:24:51 ERROR cannot create TUN device file node: operation not permitted
2022/02/22 04:24:51 INFO Shutdown successful

</details>

**Comment by qdm12** [comment]
> ERROR cannot create TUN device file node: operation not permitted

That's your problem, see the wiki page on tun device errors https://github.com/qdm12/gluetun/wiki/TUN-device-errors

**Comment by zara206** [comment]
I can't seem to follow the fix, I can't find anything about lxc container or /etc/pve/lxc/*.conf, I used rootless installation of docker, could that be the reasons?

**Comment by qdm12** [comment]
There are different sections in the [wiki page](https://github.com/qdm12/gluetun/wiki/TUN-device-errors), the LXC is one of them, you can try the others.

What you can try is

- `--device /dev/net/tun`
- Run with `--privileged` if that doesn't work. There might be a better way, but essentially this container needs root to fiddle with low level networking/vpn.
